Cactooz/StarWarsWiki
# Limit favorites

### Additions/changes
The following changes were implemented to save storage on Firebase:
- Added favorite limit to model, that can easily be modified. Currently set to `9`
- Made stars into components for less repetitive code
- Added toast when trying to add favorites and not being logged in
- Added toast when trying to add favorites and limit is reached
- Refactoring

### Testing
1. Start by running `yarn install`to install dependencies
2. Try adding a favorite from details view or browse view when not being logged in. Toast should show
3. Sign in and repeat the step above. Toast about not being signed in should not show
4. Now add favorites to your profile until the limit is reached
5. Try to add another one from details or browse. Toast should show saying limit is reached

### Future improvements
The toasts are limited to one. However, when clicking a star multiple times, the toasts are queued and shown sequentially. They keep showing until all are shown, or the page is refreshed/changed.
